aptos-move/block-executor/src/view.rs

```diff
@@ -545,10 +542,51 @@ impl<'a, T: Transaction, X: Executable> ParallelState<'a, T, X> {
 
         loop {
             match self.versioned_map.data().fetch_data(key, txn_idx) {
-                Ok(Versioned(version, v, layout)) => {
-                    let data_read = DataRead::Versioned(version, v.clone(), layout)
+                Ok(Versioned(version, value)) => {
+                    // If we have a known layout, upgrade RawFromStorage value to Exchanged.
+                    if let UnknownOrLayout::Known(layout) = layout {
+                        assert_matches!(
```

Comment:
Is this correct?Things can be upgraded concurrently, so assertion can no longer hold. I do not think we need it here @gelash @igor-aptos @igor-aptos